lib/logstash/pluginmanager/install.rb

```diff
@@ -56,6 +56,12 @@ def execute
     FileDependencies::Gem.hook
     options = {}
     options[:document] = []
+    if LogStash::Environment.test?
+      # More info on what do this options means
+      # https://github.com/rubygems/rubygems/blob/master/lib/rubygems/dependency_installer.rb#L56
```

Comment:
in any case the point is that the link you refer to does not include the documentation to these 2 options. you should document the intended behaviour adn why we use them here.